### ISSUE TITLE
detect passcode set

### ISSUE THREAD

**Issue by 7ynk3r** [rebeccahughes/react-native-device-info#153]
hi there,

we're looking for a way to detect if the passcode has been set for the device. we're thinking on implementing this a new library but this library also looks like a good place for it.

the idea is to create a pull request with this feature.

what do you think?

cheers,
jm

**Comment by passabilities** [comment]
Pretty good idea. You should create a PR. Looks like its pretty easy with Android

**Comment by kacynMedallia** [comment]
Opened PR for this issue:

https://github.com/rebeccahughes/react-native-device-info/pull/156

**Comment by kacynMedallia** [comment]
PRs https://github.com/rebeccahughes/react-native-device-info/pull/156 and https://github.com/rebeccahughes/react-native-device-info/pull/157 merged for this issue.  Let's close this issue @7ynk3r

**Comment by muratozgul** [comment]
"react": "15.4.2",
"react-native": "0.41.2",
"react-native-device-info": "0.10.2",

isPinOrFingerprintSet callback is not firing. I am testing on iPad Mini (iOS 9.3.5) and simulator (iOS 10.2)

`    RNDeviceInfo.isPinOrFingerprintSet(isSet => {
      console.log('isPinOrFingerprintSet callback:', isSet);
      if(isSet) {
        console.log('checkIfPasscodeSet: TRUE');
        resolve(true);
      }
      else {
        console.log('checkIfPasscodeSet: FALSE');
        resolve(false);
      }
    });`

**Comment by mongolgamestudio** [comment]
Try with NativeModules.
```
const checkIfPasscodeSet = () => {
  console.log('checkIfPasscodeSet called');
  return new Promise((resolve, reject) => {
    NativeModules.RNDeviceInfo.isPinOrFingerprintSet(isSet => {  
      console.log('isPinOrFingerprintSet callback:', isSet);  
      if(isSet) {  
        console.log('checkIfPasscodeSet: TRUE');  
        resolve(true);  
      }  
      else {  
        console.log('checkIfPasscodeSet: FALSE');  
        resolve(false);  
      }
    });
  });
};
```

**Comment by suparng-medallia** [comment]
@muratozgul Use it like this

`RNDeviceInfo.isPinOrFingerprintSet()(callback)`

The reason is that `RNDeviceInfo.isPinOrFingerprintSet()` returns the native function which accepts the callback.  `RNDeviceInfo.isPinOrFingerprintSet` itself does not do anything with the callback if you pass one.

**Comment by machour** [comment]
Implemented in 0.13.0
